### PR TITLE
feat: initial Enterprise support

### PR DESCRIPTION
**Pull request by rawkode**

### CODE REVIEW

File: charts/influxdb-enterprise/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  secret:
+    name: license
+    key: json
+```
+
+#### Shared Secret
+
+The meta cluster requires a shared internal secret to secure communication. This must be provided by specifying a secret name in the `values.yaml` file.
+
+The secret MUST container a key called `secret` that is a randomly generated string.
```
Comment (BondAnthony) [inline charts/influxdb-enterprise/README.md:44]: `contain`, auto-correct making you type container. Does this mean we should have a layout like this?

```
meta:
  sharedSecret:
    secret: aa8c346b84418626b91e1f66c14230f7
```

Comment (BondAnthony) [reply charts/influxdb-enterprise/README.md:44]: Reading further I think you need doc update.

**Comment by rawkode** [comment]
@influxdata/docs-team Could we have some help from docs before we merge this, please?

**Comment by pierwill** [comment]
Sure! Will take a look at this—

File: charts/influxdb-enterprise/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
+# InfluxDB Enterprise
+
+## QuickStart
```
Comment (pierwill) [inline charts/influxdb-enterprise/README.md:3]: ## Quick start

File: charts/influxdb-enterprise/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+This chart bootstraps an InfluxDB Enterprise cluster, with a StatefulSet for both the Meta and Data nodes.
+
+## Prerequisites
+
+- Kubernetes 1.4+
+- PV provisioner support in the underlying infrastructure (optional)
+
+### Secrets
+
+This chart REQUIRES some mandatory secrets in-order to function.
```
Comment (pierwill) [inline charts/influxdb-enterprise/README.md:23]: Maybe:

"This chart requires the following secrets in order to function:

[list or description of secrets]"

File: charts/influxdb-enterprise/templates/data-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    reporting-disabled = false
+
+    {{ if .Values.bootstrap.auth.secretName }}
+    [http]
+    auth-enabled = true
+    {{ end }}
+
+    [enterprise]
+      {{ if .Values.license.key }}
+      # license-key and license-path are mutually exclusive, use only one and leave the other blank
```
Comment (pierwill) [inline charts/influxdb-enterprise/templates/data-configmap.yaml:20]: License-key and license-path are mutually exclusive. Use only one and leave the other blank.

File: charts/influxdb-enterprise/README.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+```bash
+helm repo add influxdata https://helm.influxdata.com/
+helm upgrade --install influxdb influxdata/influxdb-enterprise --namespace monitoring
+```
+
+> **Tip**: `helm upgrade --install [RELEASE] [CHART] [FLAGS]` can be shortened : `helm upgrade -i [RELEASE] [CHART] [FLAGS]`
+
+## Introduction
+
+This chart bootstraps an InfluxDB Enterprise cluster, with a StatefulSet for both the Meta and Data nodes.
```
Comment (pierwill) [inline charts/influxdb-enterprise/README.md:14]: We generally do not capitalize "meta" or "data" in "meta node".

**Pull request merged**